hadoop-common-project/hadoop-common/src/main/proto/RpcHeader.proto

```diff
@@ -91,6 +91,7 @@ message RpcRequestHeaderProto { // the header for the RpcRequest
   optional RPCTraceInfoProto traceInfo = 6; // tracing info
   optional RPCCallerContextProto callerContext = 7; // call context
   optional int64 stateId = 8; // The last seen Global State ID
+  map<string, int64> nameserviceStateIds = 9; // Last seen state IDs for multiple nameservices.
```

Comment:
Should we not use map in proto?
It is not compatibility to old hadoop version which is based on protobuf2.5.

Reply:
On my laptop this compiles okay with protoc-2.5.0.
https://developers.google.com/protocol-buffers/docs/proto#maps

The protobuf v2 spec also show the map syntax.

Reply:
I'm moving the protocol changes to HDFS-16669

Reply:
But it is not supported in my Mac. My proto compile version as below:
```
 % protoc --version
libprotoc 2.5.0
```
In github release,I found it is supported after v3.0.0-alpha-2. See link: https://github.com/protocolbuffers/protobuf/releases?page=10

Reply:
Interesting. I'm also on mac with the same version. Anyhow, I can convert to the backward compatible syntax.
`
message MapFieldEntry {
  key_type key = 1;
  value_type value = 2;
}
repeated MapFieldEntry map_field = N;
`

Reply:
Yeah, it does not matter. Our internal version we just used another message to replace map.